File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int answer = 0;
    public int solution(int[] numbers, int target) {
        dfs(numbers, target, 0, 0);
        return answer;
    }
    public void dfs(int[] numbers, int target, int sum, int depth) {
        if(depth==numbers.length) {
            if(target==sum) {
            answer++;
            }
        }
         else {
             int plus = sum+numbers[depth];
             int minus = sum-numbers[depth];
            dfs(numbers, target, plus, depth+1);
            dfs(numbers, target, minus, depth+1);
        }
    }
    
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let ans = 0;
    function dfs(sum, depth) {
        if(depth==numbers.length) {
            if(sum==target) {
                ans++;
            }
            return;
        }
        let plus = sum+numbers[depth];
        let minus = sum-numbers[depth];
        dfs(plus, depth+1);
        dfs(minus, depth+1);
    }
    dfs(0, 0);
    return ans;
}